web/source/text/prediction/modelManager.ts

```diff
@@ -104,7 +117,39 @@ namespace com.keyman.text.prediction {
       return !! this.registeredModels[model.id];
     }
 
-    // TODO:  actually calling this.lmEngine.predict.  Will need its own method(s).
+    /**
+     * Function     addEventListener
+     * Scope        Public
+     * @param       {string}            event     event to handle
+     * @param       {function(Event)}   func      event handler function
+     * @return      {boolean}                     value returned by util.addEventListener
+     * Description  Wrapper function to add and identify KeymanWeb-specific event handlers
+     */       
+    ['addEventListener'](event: string, func: ReadySuggestionsHandler | InvalidateSuggestionsHandler): boolean {
+      let keyman = com.keyman.singleton;
+      return keyman.util.addEventListener(ModelManager.EVENT_PREFIX + event, func);
+    }
+
+    public predict(transform: Transform, context: Context) {
+      // If there's no active model, there can be no predictions.
+      if(!this.currentModel) {
+        return;
+      }
+
+      let keyman = com.keyman.singleton;
+      var promise = this.currentPromise = this.lmEngine.predict(transform, context);
+
+      // We've already invalidated any suggestions resulting from any previously-existing Promise -
+      // may as well officially invalidate them via event.
+      keyman.util.callEvent(ModelManager.EVENT_PREFIX + "invalidate", null);
+
+      let mm = this;
+      promise.then(function(suggestions: Suggestion[]) {
+        if(promise == mm.currentPromise) {
+          keyman.util.callEvent(ModelManager.EVENT_PREFIX + "suggestions_ready", suggestions);
```

Comment:
The convention in kmwkeyboards.ts is lower-case, no underscores `suggestionsready`

Maybe l.144 becomes `invalidatesuggestions`